includes/Classifai/Providers/AWS/AmazonPolly.php

```diff
@@ -0,0 +1,595 @@
+				$new_secret_access_key !== $settings[ static::ID ]['secret_access_key'] ||
+				$new_aws_region !== $settings[ static::ID ]['aws_region']
+			) {
+				$is_credentials_changed = true;
+			}
+
+			if ( $is_credentials_changed ) {
+				$new_settings[ static::ID ]['access_key_id']     = $new_access_key_id;
+				$new_settings[ static::ID ]['secret_access_key'] = $new_secret_access_key;
+				$new_settings[ static::ID ]['aws_region']        = $new_aws_region;
+				$new_settings[ static::ID ]['voices']            = $this->connect_to_service(
+					array(
+						'access_key_id'     => $new_access_key_id,
+						'secret_access_key' => $new_secret_access_key,
+						'aws_region'        => $new_aws_region,
+					)
+				);
+
+				if ( ! empty( $new_settings[ static::ID ]['voices'] ) ) {
+					$new_settings[ static::ID ]['authenticated'] = true;
+				} else {
+					$new_settings[ static::ID ]['voices']        = [];
+					$new_settings[ static::ID ]['authenticated'] = false;
+				}
+			}
+		} else {
+			$new_settings[ static::ID ]['access_key_id']     = $settings[ static::ID ]['access_key_id'];
+			$new_settings[ static::ID ]['secret_access_key'] = $settings[ static::ID ]['secret_access_key'];
+			$new_settings[ static::ID ]['aws_region']        = $settings[ static::ID ]['aws_region'];
+
+			add_settings_error(
+				$this->feature_instance->get_option_name(),
+				'classifai-ams-polly-auth-empty',
+				esc_html__( 'One or more credentials required to connect to the Amazon Polly service is empty.', 'classifai' ),
+				'error'
+			);
+		}
+
+		$new_settings[ static::ID ]['voice'] = sanitize_text_field( $new_settings[ static::ID ]['voice'] ?? $settings[ static::ID ]['voice'] );
+
+		return $new_settings;
+	}
+
+	/**
+	 * Connects to the Amazon Polly service.
+	 *
+	 * @param array $args Overridable args.
+	 * @return array
+	 */
+	public function connect_to_service( array $args = array() ): array {
+		$settings = $this->feature_instance->get_settings( static::ID );
+
+		$default = array(
+			'access_key_id'     => $settings[ static::ID ]['access_key_id'] ?? '',
+			'secret_access_key' => $settings[ static::ID ]['secret_access_key'] ?? '',
+			'aws_region'        => $settings[ static::ID ]['aws_region'] ?? 'us-east-1',
+		);
+
+		$default = wp_parse_args( $args, $default );
+
+		// Return if credentials don't exist.
+		if ( empty( $default['access_key_id'] ) || empty( $default['secret_access_key'] ) ) {
+			return array();
+		}
+
+		try {
+			/**
+			 * Filters the return value of the connect to services function.
+			 *
+			 * Returning a non-false value from the filter will short-circuit the describe voices request and return early with that value.
+			 * This filter is useful for E2E tests.
+			 *
+			 * @since 3.1.0
+			 *
+			 * @param false|mixed $response        A return value of connect to service. Default false.
+			 * @param array       $synthesize_data HTTP request arguments.
+			 */
```

Comment:
We'll want to ensure this docblock matches what JSDoc expects so it ends up in our documentation. See the `classifai_aws_polly_synthesize_speech_args` filter for an example